components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -694,26 +693,36 @@ setup_file (      IOFileSpecs& filespecs,
   m_resume_output_file = false;
 }
 /*===============================================================================================*/
-void set_file_header(const std::string& filename)
+void OutputManager::set_file_header(const std::string& filename)
 {
   using namespace scorpio;
 
   // TODO: All attributes marked TODO below need to be set.  Hopefully by a universal value that reflects
   // what the attribute is.  For example, git-hash should be the git-hash associated with this version of
   // the code at build time for this executable.
+  auto& p = m_params.sublist("provenance");
+  auto now = std::chrono::system_clock::now();
+  std::time_t time = std::chrono::system_clock::to_time_t(now);
+  std::stringstream timestamp;
+  timestamp << "created on " << std::ctime(&time);
+  std::string ts_str = timestamp.str();
+  ts_str = std::strtok(&ts_str[0],"\n"); // Remove the \n appended by ctime
+
   set_attribute<std::string>(filename,"source","E3SM Atmosphere Model Version 4 (EAMxx)");  // TODO: probably want to make sure that new versions are reflected here.
-  set_attribute<std::string>(filename,"case","");  // TODO
+  set_attribute<std::string>(filename,"case",p.get<std::string>("caseid","NONE"));  // TODO
   set_attribute<std::string>(filename,"title","EAMxx History File");
-  set_attribute<std::string>(filename,"compset","");  // TODO
-  set_attribute<std::string>(filename,"git_hash","");  // TODO
-  set_attribute<std::string>(filename,"host","");  // TODO
-  set_attribute<std::string>(filename,"version","");  // TODO
-  set_attribute<std::string>(filename,"initial_file","");  // TODO
-  set_attribute<std::string>(filename,"topography_file","");  // TODO
-  set_attribute<std::string>(filename,"contact","");  // TODO
-  set_attribute<std::string>(filename,"institution_id","");  // TODO
-  set_attribute<std::string>(filename,"product","");  // TODO
-  set_attribute<std::string>(filename,"component","ATM");
+  set_attribute<std::string>(filename,"source","E3SM Atmosphere Model (EAMxx)");
+  set_attribute<std::string>(filename,"eamxx_version",EAMXX_VERSION);
+  set_attribute<std::string>(filename,"git_version",p.get<std::string>("git_version",EAMXX_GIT_VERSION));
+  set_attribute<std::string>(filename,"hostname",p.get<std::string>("hostname","UNKNOWN"));
+  set_attribute<std::string>(filename,"username",p.get<std::string>("username","UNKNOWN"));
+  set_attribute<std::string>(filename,"atm_initial_conditions_file",p.get<std::string>("initial_conditions_file","NONE"));
+  set_attribute<std::string>(filename,"topography_file",p.get<std::string>("topography_file","NONE"));
+  set_attribute<std::string>(filename,"contact","e3sm-data-support@listserv.llnl.gov");
+  set_attribute<std::string>(filename,"institution_id","E3SM-Projet");
+  set_attribute<std::string>(filename,"product",(m_is_model_restart_output ? "model-restart" : "model-output"));  // TODO
```

Comment:
Is this still "TODO", I see the "TODO" comment at the end of line is still there.